### PR TITLE
add `mul!` for `AbstractTriangular`

### PR DESCRIPTION
**Pull request by mzgubic**
Solves https://github.com/jump-dev/MutableArithmetics.jl/issues/152

I looked at https://github.com/jump-dev/MutableArithmetics.jl/pull/89/files for inspiration but not sure if I got it right.

### CODE REVIEW

**Comment by mzgubic** [comment]
Tests pass locally

File: src/dispatch.jl [hunk 1]
```diff
@@ -167,6 +167,14 @@ function LinearAlgebra.mul!(
     return _mul!(ret, A, B)
 end
 
+function LinearAlgebra.mul!(
+    ret::AbstractVector{<:AbstractMutable},
+    A::LinearAlgebra.AbstractTriangular,
+    B::AbstractVector
```
Comment (odow) [inline src/dispatch.jl:173]:
```suggestion
    B::AbstractVector,
```

File: src/dispatch.jl [hunk 2]
```diff
@@ -183,6 +191,14 @@ function LinearAlgebra.mul!(
     return _mul!(ret, A, B)
 end
 
+function LinearAlgebra.mul!(
+    ret::AbstractMatrix{<:AbstractMutable},
+    A::LinearAlgebra.AbstractTriangular,
+    B::AbstractMatrix
```
Comment (odow) [inline src/dispatch.jl:197]:
```suggestion
    B::AbstractMatrix,
```

File: src/Test/array.jl [hunk 3]
```diff
@@ -575,6 +575,7 @@ function triangular_test(x)
             @test MA.iszero!!(y[i, j])
         end
     end
+
```
Comment (odow) [inline src/Test/array.jl:578]:
```suggestion
```

**Comment by odow** [comment]
Yeah, getting all of these dispatches correct is pretty painful. It's largely been whack-a-mole as ambiguity errors crop up.

**Comment by mzgubic** [comment]
Thanks for taking a look and merging so quickly

**Pull request merged**